Dockerfile

```diff
@@ -1,18 +1,22 @@
 ARG BCI_IMAGE=registry.suse.com/bci/bci-busybox
 ARG GO_IMAGE=rancher/hardened-build-base:v1.21.8b1
 ARG ARCH="amd64"
+
+# Image that provides cross compilation tooling.
+FROM --platform=$BUILDPLATFORM rancher/mirrored-tonistiigi-xx:1.3.0 as xx
+
 FROM ${BCI_IMAGE} as bci
-FROM ${GO_IMAGE} as base-builder
+FROM --platform=$BUILDPLATFORM ${GO_IMAGE} as base-builder
+# copy xx scripts to your build stage
+COPY --from=xx / /
+RUN apk add file make git clang lld
+ARG TARGETPLATFORM
 # setup required packages
 RUN set -x && \
-    apk --no-cache add \
-    file \
-    gcc \
-    git \
-    make
+    xx-apk --no-cache add musl-dev gcc lld 
```

Comment:
don't we still need git? why lld ? Is it something to do with the cross-compilation?

Reply:
git is installed above for the BUILDPLATFORM.
Here we add only the packages for the TARGETPLATFORM, including `lld` for cross-compilation